### PR TITLE
Geomap: Optimize panel rendering behavior

### LINKED ISSUE #69774: Geomap: Optimize panel rendering
While addressing https://github.com/grafana/grafana/pull/69773 I discovered that we are currently re-rendering the entire geomap panel (note: only the react components and not the open layers div, but still) on every mouse move event. This is less than ideal. As part of this we are also re-rendering the Geomap Tooltip component on every mousemove leading to undesirable tooltip UX.

### PR DESCRIPTION
**Pull request by nmarrs**
Prevents rendering geomap panel react components (tooltip, legend, etc. not open layers UI) on every mouse move event. Now panel will only re-render when a tooltip is actually displayed. This. design could be improved further by only re-rendering the `GeomapTooltip` component vs. the entire `GeomapPanel` component tree but this is "good enough" for now. It seems that these changes fix the laggy UX seen [in this PR](https://github.com/grafana/grafana/pull/69773) :)

Before

https://github.com/grafana/grafana/assets/22381771/a55b0d84-48a8-416d-8e82-eec76def612a

After

https://github.com/grafana/grafana/assets/22381771/0be1f182-5bf5-4ef8-a263-4a0083e3bb9a

Fixes https://github.com/grafana/grafana/issues/69774

### CODE REVIEW

**Comment by ryantxu** [comment]
Not that the panel does not actually render with each react update -- openlayers will only render when we send it events

File: public/app/plugins/panel/geomap/utils/tootltip.ts [hunk 1]
```diff
@@ -96,7 +96,10 @@ export const pointerMoveListener = (evt: MapBrowserEvent<MouseEvent>, panel: Geo
   panel.hoverPayload.layers = layers.length ? layers : undefined;
   panel.props.eventBus.publish(panel.hoverEvent);
 
-  panel.setState({ ttip: { ...hoverPayload } });
+  // This check optimizes Geomap panel re-render behavior (without it, Geomap renders on every mouse move event)
+  if (panel.state.ttip === undefined || panel.state.ttip?.layers !== hoverPayload.layers || hoverPayload.layers) {
```
Comment (drew08t) [inline public/app/plugins/panel/geomap/utils/tootltip.ts:100]: Does this catch the edge case where the mouse can move directly from one tooltip to another?

Comment (nmarrs) [reply public/app/plugins/panel/geomap/utils/tootltip.ts:100]: Not 100% sure though in that case the layers object should be different so it should trigger a re-render 🤷🏻‍♂️

Comment (nmarrs) [reply public/app/plugins/panel/geomap/utils/tootltip.ts:103]: Yes it appears to work great with for this edge case :) 

https://github.com/grafana/grafana/assets/22381771/1170359d-cdcc-4125-8828-800b52a863dc

**Review by drew08t — APPROVED** [review]

**Pull request merged**